### PR TITLE
chore: Add Ruby 3.0 to CI

### LINKED ISSUE #533: Ruby 3.0 support
Ruby 3.0 is expected to be released tomorrow. We should add it to CI as soon as possible.

One significant change for us is that `Random::DEFAULT` is deprecated and now refers to the `Random` class instead of a `Random` instance. We call `Random::DEFAULT.bytes` in `Trace.generate_span_id` and `Trace.generate_trace_id`. This will continue to be valid until (presumably) Ruby 3.1, at which time we will need to call `Random::bytes` instead. Unfortunately, Ruby 2.5 does not provide a `Random::bytes` class method.

We will drop support for Ruby 2.5 when it is deprecated at the end of March 2021. At that time, we can update the ID generation code.

### PR DESCRIPTION
**Pull request by dazuma**
Closes #533 

Caveats:

* Currently, the Ruby 3.0 test omits the otlp exporter, because google-protobuf is not yet compatible with Ruby 3.
* The Ruby 3.0 test also omits versions of ruby-kafka older than 1.3 because of an unexplained test failure. It's possible the libraries are incompatible with Ruby 3, but I'd like to get a separate opinion on that since I'm not familiar with ruby-kafka.

Notes:

* Updated Appraisals of excon, faraday, and graphql instrumentation to omit versions that run afoul of what appears to be a Ruby 3 regression (https://bugs.ruby-lang.org/issues/10499)
* Updated the Appraisal for instrumentation-rails to omit Rails 5.2 under Ruby 3 because of keyword argument compatibility.
* Updated the Appraisal of sinatra to use 2.1 instead of 2.0.7 because of Ruby 3 keyword argument compatibility.
* Fixed test cases in common and instrumentation-redis that ran afoul of Ruby 3's keyword argument change.

### CODE REVIEW

**Comment by dazuma** [comment]
@robertlaurin The ruby-kafka test "Producer tracing traces async produce calls" is timing out on Ruby 3 and I don't understand why. (I tried increasing the timeout 10x and it's still failing.) Can you take a look at it?

**Comment by dazuma** [comment]
Tests are now passing. However, see the caveats, in particular the ruby-kafka issue noted above.

**Review by fbogsany — APPROVED** [review]
Thanks @dazuma. I think we should open tracking issues to enable things blocked by Ruby 3.0.0 incompatibility, and for the Ruby Kafka timeout (Robert is away this week, and I'd rather get this in now). If you're OK with that, feel free to merge this after a rebase.

**Pull request merged**
